src/macaron/policy_engine/__main__.py

```diff
@@ -0,0 +1,178 @@
+    """Policy engine configuration."""
+
+    database_path: str
+    interactive: bool = False
+    policy_id: int | None = None
+    policy_file: str | None = None
+    show_prelude: bool = False
+
+
+global_config = Config()
+
+
+class Timer:
+    """Time an operation using context manager."""
+
+    def __init__(self, name: str) -> None:
+        self.start: float = time.perf_counter()
+        self.name: str = name
+        self.delta: float = 0.0
+        self.stop: float = 0.0
+
+    def __enter__(self) -> "Timer":
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
+        self.stop = time.perf_counter()
+        self.delta = self.stop - self.start
+        print(self.name, f"delta: {self.delta:0.4f}")
+
+
+def get_generated(database_path: os.PathLike | str) -> SouffleProgram:
+    """Get generated souffle code from database specified by configuration."""
+    metadata = MetaData()
+    engine = create_engine(f"sqlite:///{database_path}", echo=False)
+    metadata.reflect(engine)
+
+    prelude = get_souffle_import_prelude(os.path.abspath(database_path), metadata)
+
+    for table_name in metadata.tables.keys():
+        table = metadata.tables[table_name]
+        if table_name[0] == "_":
+            prelude.update(project_table_to_key(f"{table_name[1:]}_attribute", table))
+            prelude.update(project_with_fk_join(table))
+
+    return prelude
+
+
+def copy_prelude(database_path: os.PathLike | str, sfl: SouffleWrapper, prelude: SouffleProgram | None = None) -> None:
+    """
+    Generate and copy the prelude into the souffle instance's include directory.
+
+    Parameters
+    ----------
+    database_path: os.PathLike | str
+        The path to the database the facts will be imported from
+    sfl: SouffleWrapper
+        The souffle execution context object
+    prelude: SouffleProgram | None
+        Optional, the prelude to use for the souffle program, if none is given the default prelude is generated from
+        the database at database_path.
+    """
+    if prelude is None:
+        prelude = get_generated(database_path)
+    sfl.copy_to_includes("import_data.dl", str(prelude))
+
+    folder = os.path.join(os.path.dirname(__file__), "prelude")
+    for file_name in os.listdir(folder):
+        full_file_name = os.path.join(folder, file_name)
+        if not os.path.isfile(full_file_name):
+            continue
+        with open(full_file_name, encoding="utf-8") as file:
+            text = file.read()
+            sfl.copy_to_includes(file_name, text)
+
+
+def policy_engine(config: type[Config], policy_file: str) -> dict:
+    """Invoke souffle and report result."""
+    with SouffleWrapper() as sfl:
+        copy_prelude(config.database_path, sfl)
+        with open(policy_file, encoding="utf-8") as file:
+            text = file.read()
+
+        try:
+            res = sfl.interpret_text(text)
+        except SouffleError as error:
+            print(error.command)
```

Comment:
Please use logger and avoid `print`. Please check other uses of `print` too.

Reply:
Fixed in https://github.com/oracle-samples/macaron/pull/46/commits/ee565267d152c2e1439bec8ed96c0730d4fc8f6d